lib/widgets/message_list.dart

```diff
@@ -442,6 +445,14 @@ const _kShortMessageHeight = 80;
 // previous batch.
 const kFetchMessagesBufferPixels = (kMessageListFetchBatchSize / 2) * _kShortMessageHeight;
 
+/// The message list.
+///
+/// Takes the full screen width,
+/// keeping the message list and scroll-to-bottom button
+/// out of the horizontal insets with transparent [SafeArea] padding.
+/// When there is no [ComposeBox],
+/// adds transparent [SliverSafeArea] and [SafeArea] padding
+/// to the message-list content and the scroll-to-bottom button, respectively.
```

Comment:
This feels a bit too specific about how it does what it does, in the sort of way that's at risk of becoming wrong over time. How about this?
```suggestion
/// Takes the full screen width, keeping its contents
/// out of the horizontal insets with transparent [SafeArea] padding.
/// When there is no [ComposeBox], also takes responsibility
/// for dealing with the bottom inset.
```